master/internal/stream/system.go

```diff
@@ -364,6 +344,42 @@ func prepareWebsocketMessage(obj stream.PreparableMessage) interface{} {
 	return msg
 }
 
+// startupMsgHandler reads and responds to startup messages sent by streaming clients.
+func startupMsgHandler(
+	ctx context.Context,
+	startupMsg StartupMsg,
+	ps *PublisherSet,
+	ss *SubscriptionSet,
+	user model.User,
+	streamer *stream.Streamer,
+) (msgs []interface{}, err error) {
+	defer func() {
+		syncMsg := SyncMsg{SyncID: startupMsg.SyncID}
+		msgs = append(msgs, syncMsg.toPreparedMessage())
+	}()
```

Comment:
In general, don't defer business logic like this.  Defer is for things that need to be handled, even in errors (explicit early exits) or panics (implict early exits).  Resource freeing, removing subscriptions from a list, etc.  But if the stream breaks or is canceled or panics or whatever, don't risk emitting the "I successfully sent you a complete update!" sync message, because you didn't; you errored or panicked out.

Reply:
Sounds good, this route had made more sense when I was communicating in-band errors back to streaming clients so a failure could be distinguished from a success; however, a lack of a response works just as well